From 63085ecfdcee68a066c1079d12b5733889183783 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Tue, 16 Jan 2018 14:59:37 +0800
Subject: [PATCH 3373/5242] MLK-17380-7 usb: cdns3: add platform TPL support

commit  2a234e670e140ac234c7de9c36cb61f34fa9d904 from
https://source.codeaurora.org/external/imx/linux-imx.git

The TPL support is used to identify targeted devices during
EH2.0 and EH3.0 certification test.

Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/host.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/cdns3/host.c b/drivers/usb/cdns3/host.c
index 433bffa..4ed9733 100644
--- a/drivers/usb/cdns3/host.c
+++ b/drivers/usb/cdns3/host.c
@@ -20,6 +20,7 @@
 #include <linux/usb.h>
 #include <linux/usb/hcd.h>
 #include <linux/pm_runtime.h>
+#include <linux/usb/of.h>
 
 #include "../host/xhci.h"
 
@@ -152,6 +153,8 @@ static int cdns3_host_start(struct cdns3 *cdns)
 		ret = -ENOMEM;
 		goto err3;
 	}
+	host->hcd->tpl_support = of_usb_host_tpl_support(sysdev->of_node);
+	xhci->shared_hcd->tpl_support = host->hcd->tpl_support;
 
 	ret = usb_add_hcd(host->hcd, 0, IRQF_SHARED);
 	if (ret)
-- 
1.7.9.5

